File: src/csharp/oop/VendingMachine/Vending/States/CompleteOrderState.cs
namespace VendingMachine.Vending.States;

public class CompleteOrderState(OrderContext context) : OrderStateBase(context)
{
    public override void Process()
    {
        NextState(new IdleState(Context.VendingMachine), true);
    }

    public override void StateInfo()
    {
        Notify("Thank you for purchase");
    }
}

File: src/python/oop/VendingMachine/vending/states/complete_order_state.py
from ..order_context import OrderContext
from .order_state import OrderState
from .idle_state import IdleState


class CompleteOrderState(OrderState):
    def __init__(self, context: OrderContext):
        super().__init__(context)

    def process(self):
        self._next_state(IdleState(self.context.vm), True)

    def state_info(self):
        self.context.vm._notify(f"Thank you for purchase")
